src/utilities/tracing.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+import os
+from collections.abc import Generator
+from contextlib import contextmanager
+
+from opentelemetry import trace
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk.resources import DEPLOYMENT_ENVIRONMENT, SERVICE_NAME, Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+
+# coupling: these need to be kept in sync with semgrep-proprietary/tracing.py
+DEFAULT_TRACE_ENDPOINT = "https://telemetry.semgrep.dev/v1/traces"
+DEFAULT_DEV_ENDPOINT = "https://telemetry.dev2.semgrep.dev/v1/traces"
+DEFAULT_LOCAL_ENDPOINT = "http://localhost:4318/v1/traces"
+
+MCP_SERVICE_NAME = "mcp"
+
+
+def get_trace_endpoint() -> tuple[str, str]:
+    """Get the appropriate trace endpoint based on environment."""
+    env = os.environ.get("ENVIRONMENT", "dev").lower()
+
+    if env == "prod":
+        return (DEFAULT_TRACE_ENDPOINT, "prod")
+    elif env == "local":
+        return (DEFAULT_LOCAL_ENDPOINT, "local")
+    else:
+        return (DEFAULT_DEV_ENDPOINT, "dev")
```

Comment:
I'd suggest we get these values from env variables. We can define 2 variables here: 

- [OTEL_EXPORTER_OTLP_TRACES_ENDPOINT](https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/)
- ENVIRONMENT

This way we can avoid hard-coding URLs in the code and users will be able to send traces to their own observability cluster.

Reply:
Makes sense, yes. Let's add the `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` env var as that's a community standard for opentelemetry. 

I  think we should deprecate the other env variables, I understand that you may want to first remove it in other places first.